presto-spi/src/main/java/com/facebook/presto/spi/statistics/Estimate.java

```diff
@@ -25,27 +26,38 @@
     //      Skipping for now as there hard to compute it properly and so far we do not have
     //      usecase for that.
 
-    private static final Estimate UNKNOWN = new Estimate(Double.NaN);
+    private static final Estimate UNKNOWN = new Estimate(NaN);
     private static final Estimate ZERO = new Estimate(0);
 
     private final double value;
 
-    public static Estimate unknownValue()
+    public static Estimate unknown()
     {
         return UNKNOWN;
     }
 
-    public static Estimate zeroValue()
+    public static Estimate zero()
     {
         return ZERO;
     }
 
-    public Estimate(double value)
+    public static Estimate create(double value)
```

Comment:
s/create/estimate

then you could static import this

Reply:
Although i agree that `createObjectName` sounds more fluent that `ObectName.create`, i can see some disadvantages in the `createObjectName` approach. Here some of them:

- Additional (static) import introduces additional merge conflicts
- It is not always clear what is the exact name of the factory method. For example taking a class called `DummyFunnyCircle`, it is not always obvious how's the factory method is called. `createDummyFunnyCircle`? `createFunnyCircle`? `createCircle`?
- Static import may introduce a name clash. For example in test classes you may want to have a private factory method called `createFunnyCircle(some parameters)`. In that method you have no other way, but use `DummyFunnyCircle.createFunnyCircle()` without static import. Using it without static import is tedious, and tautological.

I can agree that these arguments are not very strong. But still, i think it is better to follow some existing conventions:

- https://docs.oracle.com/javase/tutorial/datetime/overview/naming.html
- http://www.informit.com/articles/article.aspx?p=1216151 (check the last paragraph)

Considering that we are broadly using  `Optional` and Guava's `ImmutableList`, `ImmutableMap`, `Immutable*...` i think that it is more consistent to use similar convention when naming the factory methods. Just because people are more used to it.

I'm going to rename `Estimate.create` to `Estimate.of` to follow the `Optional` convention.

Also i think we should adopt this convention all across the code.

For example:

- Some neutral, "empty" value must be expressed as `Something.empty()`, and not `EMPTY_SOMETHING`, `SOMETHING_EMPTY`, `emptySomething()`, so on.
- Static factory methods should follow this convention:  https://docs.oracle.com/javase/tutorial/datetime/overview/naming.html
- Builder constructors should look like `Something.builder()`, and not `bulderOfSomething` or similar.

Ideally it would be nice to have a static rule that checks it. But i don't think it is easily implementable.

CC: @electrum @martint